dynamic_programming/subset_generation.py

```diff
@@ -1,44 +1,44 @@
-# Print all subset combinations of n element in given set of r element.
+# Return all subset combinations of n element in given set of r element.
 
 
 def combination_util(arr, n, r, index, data, i):
     """
-    Current combination is ready to be printed, print it
+    Current combination is ready to be returned, return it
     arr[]  ---> Input Array
     data[] ---> Temporary array to store current combination
     start & end ---> Staring and Ending indexes in arr[]
     index  ---> Current index in data[]
-    r ---> Size of a combination to be printed
+    r ---> Size of a combination to be returned
     """
     if index == r:
-        for j in range(r):
-            print(data[j], end=" ")
-        print(" ")
-        return
+        return tuple(data)
     #  When no more elements are there to put in data[]
     if i >= n:
-        return
+        return None
     # current is included, put next at next location
     data[index] = arr[i]
-    combination_util(arr, n, r, index + 1, data, i + 1)
+    res1 = combination_util(arr, n, r, index + 1, data, i + 1)
     # current is excluded, replace it with
     # next (Note that i+1 is passed, but
     # index is not changed)
-    combination_util(arr, n, r, index, data, i + 1)
-    # The main function that prints all combinations
-    # of size r in arr[] of size n. This function
-    # mainly uses combinationUtil()
+    res2 = combination_util(arr, n, r, index, data, i + 1)
+    if res1 is None:
+        return res2
+    elif res2 is None:
+        return res1
+    else:
+        return res1, res2
 
 
-def print_combination(arr, n, r):
+def get_combinations(arr, n, r):
```

Comment:
provide proper name with its type and description

Reply:
i changed the code check it once